File: chisel_py/src/chisel/checker/controllers/skill_setup_controller.py
from dataclasses import dataclass
from pathlib import Path

from chisel.checker.models.agent_skill import SkillSetupResult, SkillTarget
from chisel.checker.services.agent_skills import ISkillInstaller


@dataclass(slots=True)
class SkillSetupController:
    _installer: ISkillInstaller

    def setup(
        self,
        project_path: str,
        target: SkillTarget,
        skill_names: list[str] | None = None,
        overwrite: bool = False,
        dry_run: bool = False,
    ) -> SkillSetupResult:
        results = self._installer.install(
            Path(project_path),
            target,
            skill_names=skill_names,
            overwrite=overwrite,
            dry_run=dry_run,
        )
        return SkillSetupResult(
            target=target,
            target_dir=str(self._installer.target_dir(target)),
            results=results,
        )

File: chisel_js/src/chisel/checker/controllers/skill_setup_controller.ts
import { SkillTarget, type SkillSetupResult } from "chisel/checker/models/agent_skill";
import type { ISkillInstaller } from "chisel/checker/services/agent_skills";

export class SkillSetupController {
  constructor(private readonly installer: ISkillInstaller) {}

  setup(
    projectPath: string,
    target: SkillTarget,
    options: {
      readonly skillNames?: string[];
      readonly overwrite?: boolean;
      readonly dryRun?: boolean;
    } = {},
  ): SkillSetupResult {
    return {
      target,
      targetDir: this.installer.targetDir(target),
      results: this.installer.install(projectPath, target, options),
    };
  }
}
